src/app/_services/api/base.service.ts

```diff
@@ -89,6 +89,18 @@ export class BaseService<T extends Resource> {
       .pipe(map((data: any) => this.serializer.fromJson(data) as T));
   }
 
+  getBySampleId(id: number | string): Observable<T> {
+    // endpoint
+    let endpoint = this.endpoint.ENDPOINT;
+    if (this.endpoint.GET_BY_SAMPLE_ID) {
+      endpoint = this.endpoint.GET_BY_SAMPLE_ID;
+    }
+
+    return this.httpClient
+      .post(`${this.helperService.getApiUrl()}/${endpoint}`, [id])
+      .pipe(map((data: any) => this.serializer.fromJson(data) as T));
+  }
+
```

Comment:
this is not correct. We have a method getById in baseService. Why not use that?

Reply:
I did a separate one, as this call is a post and not a get as getById is. Also we need to pass the sampleId in the body and not in the url. So did it like this to be clear that this is another kind of call. We could add callType parameter to specify if it is post or get, but I'm not sure we want to do that. Like this it's obvious that it has nothing to do with getById.

Reply:
In base.service we have only the common methods : get all, get by id, remove, add, list. We never add specific methods here. If we have a specific method to add, we add it in the specific service. In your case, in the sample service. But you shouldn't do this now. You should use the same getById method that is beeing used for the other enteties. So please talk to Marius and Lau to create a ticket for the backend and to make the sample getById work like the rest

Reply:
It will be done when getAll will be available. It's not working from subComponent services.